cmd/internal/core/phone-home.go

```diff
@@ -152,3 +129,50 @@ func toPhoneHomeMessage(discoveryResult lldp.DiscoveryResult) *phoneHomeMessage
 	}
 	return nil
 }
+
+func (c *Core) startLLDPDiscovery(ctx context.Context, discoveryResultChan chan lldp.DiscoveryResult, iface net.Interface) {
+	// consider only switch port interfaces
+	if !strings.HasPrefix(iface.Name, "swp") {
```

Comment:
In cmd/internal/core/detect-changes.go line 32 the interface is checked for "swp" prefix.
If the same check gets kept in cmd/internal/core/phone-home.go then behavious is consistent across callers (both only call this function for "swp" interfaces) and the duplicate check here could be dropped.

Reply:
Yes it is a bit too spread, will think of a solution, maybe factoring interface listing to a separate func (which also detects switch os in the future, SONiC hast "Ethernet.." )

Reply:
Yes